### ISSUE TITLE
No training with AWQ QLORA (Qwen 2.5 VL)

### ISSUE THREAD

**Issue by ma7555** [modelscope/ms-swift#3698]
**Describe the bug**

0 loss when finetuning AWQ model with QLORA

```
CUDA_VISIBLE_DEVICES=0 \
swift sft \
    --model Qwen/Qwen2.5-VL-3B-Instruct-AWQ \
    --train_type lora \
    --dataset 'AI-ModelScope/alpaca-gpt4-data-en#500' \
    --torch_dtype float16 \
    --num_train_epochs 1 \
    --per_device_train_batch_size 1 \
    --per_device_eval_batch_size 1 \
    --learning_rate 1e-4 \
    --lora_rank 8 \
    --lora_alpha 32 \
    --target_modules all-linear \
    --gradient_accumulation_steps 16 \
    --eval_steps 50 \
    --save_steps 50 \
    --save_total_limit 5 \
    --logging_steps 5 \
    --max_length 2048 \
    --output_dir output \
    --system 'You are a helpful assistant.' \
    --warmup_ratio 0.05 \
    --dataloader_num_workers 4 \
    --model_author swift \
    --model_name swift-robot
```
```
{'loss': 0.0, 'token_acc': 0.0, 'grad_norm': nan, 'learning_rate': 0.0, 'memory(GiB)': 5.75, 'train_speed(iter/s)': 0.087389, 'epoch': 0.03, 'global_step/max_steps': '1/30', 'percentage': '3.33%', 'elapsed_time': '11s', 'remaining_time': '5m 24s'}
```                                                     

**Your hardware and system info**
Single 4080 GPU running on WSL2
```
absl-py                       2.1.0
accelerate                    1.4.0
addict                        2.4.0
aiofiles                      23.2.1
aiohappyeyeballs              2.5.0
aiohttp                       3.11.13
aiosignal                     1.3.2
aliyun-python-sdk-core        2.16.0
aliyun-python-sdk-kms         2.16.5
annotated-types               0.7.0
anyio                         4.8.0
attrdict                      2.0.1
attrs                         25.1.0
autoawq                       0.2.7.post3
av                            14.2.0
binpacking                    1.5.2
bitsandbytes                  0.45.4
certifi                       2025.1.31
cffi                          1.17.1
charset-normalizer            3.4.1
click                         8.1.8
contourpy                     1.3.1
cpm-kernels                   1.0.11
crcmod                        1.7
cryptography                  44.0.2
cycler                        0.12.1
dacite                        1.9.2
datasets                      3.2.0
decorator                     4.4.2
decord                        0.6.0
dill                          0.3.8
distro                        1.9.0
einops                        0.8.1
fastapi                       0.115.11
ffmpy                         0.5.0
filelock                      3.17.0
flash_attn                    2.7.4.post1
fonttools                     4.56.0
frozenlist                    1.5.0
fsspec                        2024.9.0
future                        1.0.0
gradio                        5.20.1
gradio_client                 1.7.2
groovy                        0.1.2
grpcio                        1.71.0
h11                           0.14.0
httpcore                      1.0.7
httpx                         0.28.1
huggingface-hub               0.29.3
idna                          3.10
imageio                       2.37.0
imageio-ffmpeg                0.6.0
importlib_metadata            8.6.1
jieba                         0.42.1
Jinja2                        3.1.6
jiter                         0.9.0
jmespath                      0.10.0
joblib                        1.4.2
kiwisolver                    1.4.8
Markdown                      3.7
markdown-it-py                3.0.0
MarkupSafe                    2.1.5
matplotlib                    3.10.1
mdurl                         0.1.2
modelscope                    1.23.2
moviepy                       1.0.3
mpmath                        1.3.0
ms_swift                      3.3.0.dev0   /mnt/c/Users/ma7mo/Documents/GitHub/ms-swift
multidict                     6.1.0
multiprocess                  0.70.16
networkx                      3.4.2
nltk                          3.9.1
numpy                         1.26.4
nvidia-cublas-cu12            12.4.5.8
nvidia-cuda-cupti-cu12        12.4.127
nvidia-cuda-nvrtc-cu12        12.4.127
nvidia-cuda-runtime-cu12      12.4.127
nvidia-cudnn-cu12             9.1.0.70
nvidia-cufft-cu12             11.2.1.3
nvidia-curand-cu12            10.3.5.147
nvidia-cusolver-cu12          11.6.1.9
nvidia-cusparse-cu12          12.3.1.170
nvidia-cusparselt-cu12        0.6.2
nvidia-nccl-cu12              2.21.5
nvidia-nvjitlink-cu12         12.4.127
nvidia-nvtx-cu12              12.4.127
openai                        1.66.1
orjson                        3.10.15
oss2                          2.19.1
packaging                     24.2
pandas                        2.2.3
peft                          0.14.0
pillow                        11.1.0
pip                           25.0.1
proglog                       0.1.10
propcache                     0.3.0
protobuf                      6.30.0
psutil                        7.0.0
pyarrow                       19.0.1
pycparser                     2.22
pycryptodome                  3.21.0
pydantic                      2.10.6
pydantic_core                 2.27.2
pydub                         0.25.1
Pygments                      2.19.1
pyparsing                     3.2.1
python-dateutil               2.9.0.post0
python-multipart              0.0.20
pytz                          2025.1
PyYAML                        6.0.2
qwen-vl-utils                 0.0.10
regex                         2024.11.6
requests                      2.32.3
rich                          13.9.4
rouge                         1.0.1
ruff                          0.9.10
safehttpx                     0.1.6
safetensors                   0.5.3
scipy                         1.15.2
semantic-version              2.10.0
sentencepiece                 0.2.0
setuptools                    69.5.1
shellingham                   1.5.4
simplejson                    3.20.1
six                           1.17.0
sniffio                       1.3.1
sortedcontainers              2.4.0
starlette                     0.46.1
sympy                         1.13.1
tensorboard                   2.19.0
tensorboard-data-server       0.7.2
tiktoken                      0.9.0
timm                          1.0.15
tokenizers                    0.21.0
tomlkit                       0.13.2
torch                         2.6.0
torchaudio                    2.6.0+cu124
torchvision                   0.21.0+cu124
tqdm                          4.67.1
transformers                  4.50.2
transformers-stream-generator 0.0.5
triton                        3.2.0
trl                           0.15.2
typer                         0.15.2
typing_extensions             4.12.2
tzdata                        2025.1
urllib3                       2.3.0
uvicorn                       0.34.0
websockets                    15.0.1
Werkzeug                      3.1.3
wheel                         0.45.1
xxhash                        3.5.0
yarl                          1.18.3
zipp                          3.21.0
zstandard                     0.23.0
```

**Comment by TheGreatLy** [comment]
me too

**Comment by Jintao-Huang** [comment]
<img width="855" alt="Image" src="https://github.com/user-attachments/assets/18449477-0380-4ff6-98c8-202691aacdce" />

The issue has not been reproduced. It looks like the loss being zero is caused by NaNs.
